include/boost/process/v2/detail/impl/exe.ipp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2022 Klemens D. Morgenstern
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+#ifndef BOOST_PROCESS_V2_IMPL_DETAIL_EXE_IPP
+#define BOOST_PROCESS_V2_IMPL_DETAIL_EXE_IPP
+
+#include <boost/process/v2/detail/exe.hpp>
+#include <boost/process/v2/detail/config.hpp>
+#include <boost/process/v2/detail/last_error.hpp>
+#include <boost/process/v2/detail/throw_error.hpp>
+
+BOOST_PROCESS_V2_BEGIN_NAMESPACE
+
+namespace detail
+{
+
+namespace ext
+{
+
+#if defined(_WIN32)
+
+HANDLE open_process_with_debug_privilege(pid_type pid, error_code & ec) {
+    HANDLE proc = nullptr;
+    HANDLE hToken = nullptr;
+    LUID luid;
+    TOKEN_PRIVILEGES tkp;
+    if (OpenProcessToken(GetCurrentProcess(), TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY, &hToken))
+    {
+        if (LookupPrivilegeValue(nullptr, SE_DEBUG_NAME, &luid))
+        {
+            tkp.PrivilegeCount = 1;
+            tkp.Privileges[0].Luid = luid;
+            tkp.Privileges[0].Attributes = SE_PRIVILEGE_ENABLED;
+            if (AdjustTokenPrivileges(hToken, false, &tkp, sizeof(tkp), nullptr, nullptr))
+            {
+                proc = OpenProcess(PROCESS_ALL_ACCESS, false, pid);
+            }
+        }
+        CloseHandle(hToken);
+    }
+    if (!proc)
+        proc = OpenProcess(PROCESS_ALL_ACCESS, false, pid);
+    if (!proc)
+        ec = detail::get_last_error();
+    return proc;
+}
+
+BOOL is_x86_process(HANDLE proc, error_code & ec) {
```

Comment:
Just out of curiosity: Is there something like that on linux?

Reply:
If linux has anything like this it is bound to be provided by the procfs. A quick google search didn't give me any results i was hoping for, but i can look more after addressing more pressing things you critiqued :)